### PR TITLE
3d physics collider

### PR DESCRIPTION
**Pull request by super626**
1. new class Physics3DCollider
2. trigger test using Physics3DCollider
3. Physics3DCollider is used instead of RigidBody in physics demo of terrain, because Physics3DCollider has better performance than rigid body for static objects.

### CODE REVIEW

**Comment by pandamicro** [comment]
@super626 Please assign someone to review your PR

**Comment by super626** [comment]
@masterhou 
@dabingnn 
Review please.

File: cocos/physics3d/CCPhysics3DObject.cpp [hunk 1]
```diff
@@ -370,6 +370,190 @@ bool Physics3DRigidBody::isKinematic() const
...
+{
+    return _btGhostObject->getRestitution();
+}
+
+void Physics3DCollider::setRestitution(float rest)
+{
+    _btGhostObject->setRestitution(rest);
+}
+
+bool Physics3DCollider::IsTrigger() const
```
Comment (pandamicro) [inline cocos/physics3d/CCPhysics3DObject.cpp]: This should be `isTrigger`, first letter of function must be lower case

File: cocos/physics3d/CCPhysics3DObject.h [hunk 2]
```diff
@@ -358,6 +360,106 @@ class CC_DLL Physics3DRigidBody : public Physics3DObject
...
+
+    static Physics3DCollider* create(Physics3DColliderDes *info);
+
+    btGhostObject* getGhostObject() const { return _btGhostObject; }
+
+    /** Set trigger. */
+    void setTrigger(bool isTrigger);
+
+    /** Check is a trigger. */
+    bool IsTrigger() const;
```
Comment (pandamicro) [inline cocos/physics3d/CCPhysics3DObject.h]: should be `isTrigger`

**Pull request merged**